SCons/ActionTests.py

```diff
@@ -2255,7 +2255,11 @@ def func1(a, b, c):
             (3, 11): (bytearray(b'3, 3, 0, 0,(),(),(\x97\x00|\x00S\x00),(),()'),),
             (3, 12): (bytearray(b'3, 3, 0, 0,(),(),(\x97\x00|\x00S\x00),(),()'),),
             (3, 13): (bytearray(b'3, 3, 0, 0,(),(),(\x95\x00U\x00$\x00),(),()'),),
-            (3, 14): (bytearray(b'3, 3, 0, 0,(),(),(\x80\x00R\x00"\x00),(),()'),),
+            (3, 14): (bytearray(
```

Comment:
You can skip the if platform logic, and just have multiple per version goldens like above. Just add a comment that it's win32..

Reply:
Was actually hoping to eventually ease us out of having the tuples in that one dict which requires using `assertIn`. But can do it the other way, too.

Reply:
The tuples became necessary because some minor version in the past introduced a change in the hash generated. So I don't think we should try to eliminate it, it's necessary flexibility.

Reply:
The same type of construct as I used could take care of that, something like:
```py
    (3, 10):
        bytearray(b'3, 3, 0, 0,(N.),(),(|\x00S\x00),(),()')
        if sys.version_info[2] == 0  # 3.10.0 had unique bytecode
        else  bytearray(b'3, 3, 0, 0,(),(),(|\x00S\x00),(),()')

Reply:
I think it's only adding unnecessary complexity to this test.  The chances that we'd introduce a bug which would match the other possible value for a given version is infinitesimally small.
(And honestly I hate the  X if a else Y syntax embedded in a static data structure initialization.)